### PR TITLE
Add Queue implementation

### PR DESCRIPTION
**Pull request by Jay9596**
- Added a Generic Queue implementation with basic operations.
- Added tests for operations

### CODE REVIEW

File: src/data_structure/queue.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        self.data.push(e)
+    }
+
+    // Returns an Option with first element of the list,
+    // if it is empty returns None
+    pub fn dequeue(&mut self) -> Option<T> {
+        if self.data.is_empty() {
+            return None;
+        }
+        Some(self.data.remove(0))
```
[the remark this answers is not in the document]
Comment (Jay9596) [reply src/data_structure/queue.rs]: Ok 👍 , so should I make the changes to this, or make a new implementation of Circular Queue ?

[the remark this answers is not in the document]
Comment (AnshulMalik) [reply src/data_structure/queue.rs]: We can keep both ones :)

[the remark this answers is not in the document]
Comment (Jay9596) [reply src/data_structure/queue.rs]: I'll make a new Circular Queue implementation and add a new PR.

[the remark this answers is not in the document]
Comment (Jay9596) [reply src/data_structure/queue.rs]: Made a new PR with Circular queue implementation. #45

**Comment by Volpe95** [comment]
Closing this pull request as not all checks are satisfied and inactivity.

**Pull request closed without merging**